src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,71 @@
 #endregion
 
 using System;
+using Autofac.Features.AttributeFilters;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.IO.Events;
 using Catalyst.Abstractions.Kvm;
 using Catalyst.Abstractions.Ledger;
+using Catalyst.Abstractions.Ledger.Models;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Messaging.Correlation;
+using Catalyst.Core.Modules.Ledger.Repository;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Transaction;
+using Catalyst.Protocol.Wire;
+using Nethermind.Core.Crypto;
 using Nethermind.Store;
 
-namespace Catalyst.Core.Modules.Ledger 
+namespace Catalyst.Core.Modules.Ledger
 {
-    public class Web3EthApi : IWeb3EthApi
+    public class Web3EthApi : IWeb3EthApi, ITransactionReceiptResolver
     {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+        private readonly ITransactionReceiptRepository _receipts;
+        private readonly ITransactionReceivedEvent _transactionReceived;
+        private readonly IHashProvider _hashProvider;
+        private readonly PeerId _peerId;
+
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, IDeltaExecutor executor, IStorageProvider storageProvider, IStateProvider stateProvider, ITransactionReceiptRepository receipts, ITransactionReceivedEvent transactionReceived, IHashProvider hashProvider, IPeerSettings peerSettings)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            _receipts = receipts;
+            _transactionReceived = transactionReceived ?? throw new ArgumentNullException(nameof(transactionReceived));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)